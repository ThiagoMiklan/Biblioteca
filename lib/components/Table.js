"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _type_validations = require("../tools/type_validations");

var style_cursor = {
  cursor: "pointer"
};
var props_obj = {
  itens_body: "array",
  itens_header: "array",
  itens_footer: "array",
  definition: "string",
  onClickRow: "function",
  header_definition: "string"
};

var Table = function Table(props) {
  (0, _type_validations.validate)(props, props_obj, "Table");
  return assembleTable(props);
};

function assembleTable(props) {
  var definition = props.definition == undefined ? "table" : "table " + props.definition;

  var code = /*#__PURE__*/_react["default"].createElement("table", {
    className: definition
  }, isArrayOk(props.itens_header) && assembleHeader(props), isArrayOk(props.itens_body) && assembleBody(props.itens_body, props.onClickRow), isArrayOk(props.itens_footer) && assembleFooter(props.itens_footer));

  return code;
}

function isArrayOk(array) {
  var output = false;

  if (Array.isArray(array) && array != undefined) {
    if (array.length > 0) {
      output = true;
    }
  }

  return output;
}

function assembleHeader(props) {
  var code_header = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null);

  var header_definition = props.header_definition;

  if (props.itens_header != undefined) {
    code_header = /*#__PURE__*/_react["default"].createElement("thead", {
      className: header_definition
    }, /*#__PURE__*/_react["default"].createElement("tr", null, props.itens_header.map(function (item_header) {
      return assembleRowHeaderFooter(item_header);
    })));
  }

  return code_header;
}

function assembleFooter(itens_footer) {
  var code_footer = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null);

  if (itens_footer != undefined) {
    code_footer = /*#__PURE__*/_react["default"].createElement("tfoot", null, /*#__PURE__*/_react["default"].createElement("tr", null, itens_footer.map(function (item_footer) {
      return assembleRowHeaderFooter(item_footer);
    })));
  }

  return code_footer;
}

function assembleBody(itens, onClickRow) {
  var code = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null);

  var getRow = function getRow(item) {
    var row = "";
    var values = Object.values(item);
    row = values.map(function (value) {
      return /*#__PURE__*/_react["default"].createElement("td", null, value);
    });
    return row;
  };

  if (itens != undefined) {
    code = /*#__PURE__*/_react["default"].createElement("tbody", null, itens.map(function (item) {
      return /*#__PURE__*/_react["default"].createElement("tr", {
        key: item["key"],
        style: style_cursor,
        onClick: onClickRow
      }, getRow(item));
    }));
  }

  return code;
}

function assembleRowHeaderFooter(item) {
  return /*#__PURE__*/_react["default"].createElement("th", null, item["value"]);
}

var _default = Table;
exports["default"] = _default;