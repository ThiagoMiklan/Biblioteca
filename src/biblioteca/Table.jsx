import React from 'react';
import PropTypes from 'prop-types';


class Table extends React.Component {


    render() {
        return (
            <div className="table">
                {assembleHeader(this.props)}
                {assembleFooter(this.props)}
                {assembleBody(this.props)}
            </div>
        );

    }
}


function assembleHeader(props) {
    var header = "";
    if (props.rows_header != undefined) {
        header =
            <thead>
                <tr>
                    {props.rows_header.map(row => assembleRow(row))}
                </tr>
            </thead>;
    }
    return header;
}

function assembleBody(props) {
    var body = "";
    if (props.rows_body != undefined) {
        body = <tbody>
                {props.rows_body.map(row_array => <tr>{row_array.map(row => assembleRowBody(row))}</tr>)}
        </tbody>;
    }
    return body;
}

function assembleFooter(props) {
    var footer = "";
    if (props.rows_footer != undefined) {
        footer = <tfoot>
            <tr>
                {props.rows_footer.map(row => assembleRow(row))}
            </tr>
        </tfoot>;
    }
    return footer;
}

function assembleRow(row) {
    if (typeof (row) != "object") {
        return <th>{row}</th>;
    } else {
        return <th><abbr title={String(row.abbr_text)}>{String(row.value)}</abbr></th>
    }
}

function assembleRowBody(row) {

    var array1 = row[0];

    if (typeof (row) != "object") {
        return <td>{row}</td>;
    } else {
        var x = row.have_link;
        if (row.have_link == true) {
        return <th><a title={row.value} href={row.href}> {row.value}</a></th>
        }
    }
}





Table.propTypes = {
    definition: PropTypes.string,
    table_footer: PropTypes.bool,
    table_header: PropTypes.bool,
    rows_header: PropTypes.array,
    rows_footer: PropTypes.array,
    rows_body: PropTypes.array
}

export default Table;