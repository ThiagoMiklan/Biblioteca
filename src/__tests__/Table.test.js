import React from 'react';
import Table from '../biblioteca/Table';
import renderer from 'react-test-renderer';

var elements_header = [
    {value:"Pos"},{value:"Team"},{value:"Pld"},{value:"W"},
    {value:"D"},{value:"L"},{value:"GF"},{value:"GA"},
    {value:"GD"},{value:"Pts"},{value:"Qualification or relegation"}
  ]
  
  var elements_footer =  elements_header;
  
  
  //2	Arsenal	38	20	11	7	65	36	+29	71	Qualification for the Champions League group stage
  var elements_body  = [
    {Pos:"2",team:"Arsenal",pld:"38",w:"20",
    D:"11",L:"7",GF:"65",GA:"36",
    GD:"+29",Pts:"71",Quali:"Qualification for the Champions League group stage"}
  ]
  

test('test table',()=>{
    const table = renderer.create(
        <Table 
           definition="bordered striped narrow fullwidth"
           itens_header={elements_header}
           itens_body={elements_body}
           itens_footer={elements_footer} />
    ).toJSON();

    expect(table).toMatchSnapshot();
})


