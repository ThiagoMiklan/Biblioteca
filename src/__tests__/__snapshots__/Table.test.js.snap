// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test table 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        <abbr
          title="Position"
        >
          Pos
        </abbr>
      </th>
      <th>
        <abbr
          title="PLayed"
        >
          Pld
        </abbr>
      </th>
      <th>
        <abbr
          title="Won"
        >
          W
        </abbr>
      </th>
      <th>
        <abbr
          title="Drawn"
        >
          D
        </abbr>
      </th>
      <th>
        <abbr
          title="Lost"
        >
          L
        </abbr>
      </th>
      <th>
        <abbr
          title="Goals for"
        >
          Gl
        </abbr>
      </th>
      <th>
        <abbr
          title="Gaols Against"
        >
          GA
        </abbr>
      </th>
      <th>
        <abbr
          title="Goal diferrence"
        >
          GD
        </abbr>
      </th>
      <th>
        <abbr
          title="Points"
        >
          PTS
        </abbr>
      </th>
      <th>
        Qualification or Relegation
      </th>
    </tr>
  </thead>
  <tfoot>
    <tr>
      <th>
        <abbr
          title="Position"
        >
          Pos
        </abbr>
      </th>
      <th>
        <abbr
          title="PLayed"
        >
          Pld
        </abbr>
      </th>
      <th>
        <abbr
          title="Won"
        >
          W
        </abbr>
      </th>
      <th>
        <abbr
          title="Drawn"
        >
          D
        </abbr>
      </th>
      <th>
        <abbr
          title="Lost"
        >
          L
        </abbr>
      </th>
      <th>
        <abbr
          title="Goals for"
        >
          Gl
        </abbr>
      </th>
      <th>
        <abbr
          title="Gaols Against"
        >
          GA
        </abbr>
      </th>
      <th>
        <abbr
          title="Goal diferrence"
        >
          GD
        </abbr>
      </th>
      <th>
        <abbr
          title="Points"
        >
          PTS
        </abbr>
      </th>
      <th>
        Qualification or Relegation
      </th>
    </tr>
  </tfoot>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <th>
        <a
          href="https://en.wikipedia.org/wiki/Manchester_United_F.C."
          title="Leicester"
        >
           
          Leicester
        </a>
      </th>
      <td>
        38
      </td>
      <td>
        23
      </td>
      <td>
        12
      </td>
      <td>
        3
      </td>
      <td>
        6
      </td>
      <td>
        36
      </td>
      <td>
        32
      </td>
      <td>
        Qualification for the Champions League Group Stage
      </td>
    </tr>
    <tr>
      <td>
        1
      </td>
      <th>
        <a
          title="Leicester"
        >
           
          Leicester
        </a>
      </th>
      <td>
        38
      </td>
      <td>
        23
      </td>
      <td>
        12
      </td>
      <td>
        3
      </td>
      <td>
        6
      </td>
      <td>
        36
      </td>
      <td>
        32
      </td>
      <td>
        Qualification for the Champions League Group Stage
      </td>
    </tr>
  </tbody>
</table>
`;
